ImperatorToCk3/Source/Imperator/ImperatorWorld.cpp

```diff
@@ -0,0 +1,108 @@
+#include "ImperatorWorld.h"
+#include "../Common/CommonFunctions.h"
+#include "../Common/Version.h"
+#include "../Configuration/Configuration.h"
+#include "Date.h"
+#include "Log.h"
+#include "OSCompatibilityLayer.h"
+#include "ParserHelpers.h"
+#include <ZipFile.h>
+#include <filesystem>
+#include <fstream>
+
+namespace fs = std::filesystem;
+
+ImperatorWorld::World::World(const Configuration& theConfiguration)
+{
+	LOG(LogLevel::Info) << "*** Hello Imperator, Roma Invicta! ***";
+	registerRegex("\\bSAV\\w*\\b", [](const std::string& unused, std::istream& theStream) {});
+	registerKeyword("version", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::singleString versionString(theStream);
+		ImperatorVersion = Version(versionString.getString());
+		Log(LogLevel::Info) << "<> Savegame version: " << versionString.getString();
+	});
+	registerKeyword("date", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::singleString dateString(theStream);
+		endDate = date(dateString.getString());
+		Log(LogLevel::Info) << "<> Date: " << dateString.getString();
+	});
+	registerKeyword("enabled_dlcs", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList dlcsList(theStream);
+		const auto& theDLCs = dlcsList.getStrings();
+		DLCs.insert(theDLCs.begin(), theDLCs.end());
+		for (const auto& dlc : DLCs) LOG(LogLevel::Info) << "<> Enabled DLC: " << dlc;
+	});
+	registerKeyword("enabled_mods", [this](const std::string& unused, std::istream& theStream) {
+		const commonItems::stringList modsList(theStream);
+		const auto& theMods = modsList.getStrings();
+		Mods.insert(theMods.begin(), theMods.end());
+		for (const auto& mod : Mods) LOG(LogLevel::Info) << "<> Enabled mod: " << mod;
+	});
+	registerKeyword("family", [this](const std::string& unused, std::istream& theStream) {
```

Comment:
I like this. I also dislike the insane method they're storing these. family inside families inside family?

PDX, why.

Reply:
Oooh. Clever.

Reply:
When I found this, I thought: "I guess they save something else family-related", but there's nothing else 😕. They redid the families heavily in some earlier patch, so maybe there was something besides families={} in family {}.